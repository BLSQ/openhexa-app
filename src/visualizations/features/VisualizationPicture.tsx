import { gql } from "@apollo/client";
import { VisualizationPicture_VisualizationFragment } from "./VisualizationPicture.generated";

type VisualizationPictureProps = {
  visualization: VisualizationPicture_VisualizationFragment;
};

const VisualizationPicture = (props: VisualizationPictureProps) => {
  const { visualization } = props;
  return (
    <div className="flex h-10 w-10 flex-shrink-0 items-center">
      <img
        className="h-10 w-10 rounded-full"
        alt=""
        src={visualization.pictureUrl}
      />
    </div>
  );
};

VisualizationPicture.fragments = {
  visualization: gql`
    fragment VisualizationPicture_visualization on ExternalDashboard {
      pictureUrl
    }
  `,
};

export default VisualizationPicture;
