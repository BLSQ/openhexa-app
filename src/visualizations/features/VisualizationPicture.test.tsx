import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import VisualizationPicture from "../features/VisualizationPicture";
import { VisualizationPicture_VisualizationFragment } from "./VisualizationPicture.generated";

describe("VisualizationPicture", () => {
  const visualization = {
    pictureUrl:
      "https://www.bluesquarehub.com/wp-content/uploads/2021/07/logosvg-white.svg",
  };
  it("renders properly", () => {
    const { container: visualizationComponent } = render(
      <VisualizationPicture visualization={visualization} />,
    );
    expect(visualizationComponent).toMatchSnapshot();
  });
});
