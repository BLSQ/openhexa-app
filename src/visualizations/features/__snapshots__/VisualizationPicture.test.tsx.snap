// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualizationPicture renders properly 1`] = `
<div>
  <div
    class="flex h-10 w-10 flex-shrink-0 items-center"
  >
    <img
      alt=""
      class="h-10 w-10 rounded-full"
      src="https://www.bluesquarehub.com/wp-content/uploads/2021/07/logosvg-white.svg"
    />
  </div>
</div>
`;
